executable-spec/src/Cardano/Ledger/Spec/STS/Update/Ideation.hs

```diff
@@ -133,90 +149,58 @@ instance HashAlgorithm hashAlgo => STS (IDEATION hashAlgo) where
 
           -- TODO: Add verification of signature inside SIPCommit
 
-          pure $! st { wssips = Map.insert (Data.commit sipc) (currentSlot) wssips
-                     , subsips = Set.insert sip subsips
+          pure $! st { wssips = wssips ⨃ [(Data.commit sipc, currentSlot)]
+                     , subsips = subsips ⨃ [(Data.commit sipc, sip)]
                      }
 
         Reveal sip -> do
           author sip ∈ dom participants ?! InvalidAuthor (author sip)
           sip ∈ subsips ?! NoSIPToReveal sip
```

Comment:
Yeah! I think so! Good catch. How does this even compiles? :scream:

Reply:
Because `Map` has a `Foldable` instance on its range :man_facepalming: 

```haskell
(∈) :: (Eq a, Foldable f) => a -> f a -> Bool
a ∈ f = elem a f
```